runtests.py

```diff
@@ -5,49 +5,57 @@
 #
 
 """
-File for running tests programmatically.
+Script for running Spyder tests programmatically.
 """
 
 # Standard library imports
 import os
 import sys
+import argparse
 
 # Third party imports
-import qtpy  # to ensure that Qt4 uses API v2
 import pytest
 
 
 # To run our slow tests only in our CIs
-run_slow = False
-if os.environ.get('CI', None) is not None or '--run-slow' in sys.argv:
-    run_slow = True
+RUN_CI = os.environ.get('CI', None) is not None
+run_slow = RUN_CI
 
-def main():
+
+def main(run_slow=run_slow, extra_args=None):
     """
-    Run pytest tests.
+    Run pytest tests for Spyder.
     """
     pytest_args = ['spyder',
                    'spyder_profiler',
-                   '-x',
                    '-vv',
                    '-rw',
-                   '--durations=10',
-                   '--cov=spyder',
-                   '--cov=spyder_profiler',
-                   '--cov-report=term-missing']
+                   '--durations=10']
 
-    if run_slow:
+    if RUN_CI:
+        pytest_args.append('-x')
+    elif extra_args:
+        pytest_args += extra_args
+    if run_slow and '--run-slow' not in pytest_args:
```

Comment:
> pytest_args is a fixed list, so there's no need to have the second condition.

Actually it isn't, as you can see from the line just before this one—the user may have added it as one of the ``extra_args``, a very likely possibility after the change you request below.

Also, ``run_slow`` != ``RUN_CI`` on ``master`` as discussed above, so unless you want to restructure how that is handled, this would still have to be there.

Reply:
Ok, then this needs to be

    if RUN_CI and '--run-slow' not in pytest_args:

Reply:
Okay sure, I did  that and ended up just refactoring the section entirely to make the logic simpler in ``3.x``. When you merge, you'll need to change it to check for the extra constant you add, or just directly, as to whether it is running on a macOS CI.